### PR TITLE
Upgrade pylint => 1.7.1

### PR DESCRIPTION
**Pull request by troydai**
Update the pylint to version 1.7.1 and fixes the newly appeared issues. Notably:

```
:no-else-return (R1705): *Unnecessary "else" after "return"*
  else following a chain of ifs, all of them containing a return statement.

:len-as-condition (C1801): *Do not use `len(SEQUENCE)` as condition value*
  Used when Pylint detects incorrect use of len(sequence) inside conditions.
  implement __reversed__, nor __getitem__ and __len__
```

In addition, as a result of the upgrade, pylint is enabled in 3.6.

### CODE REVIEW

**Comment by troydai** [comment]
I would like to merge this change as soon as possible ti reduce the impacts from other pull requests since this impact a fairly large number of files.

File: src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/custom.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def list_adla_jobs(client,
     job_list = client.list(account_name,
                            orderby="submitTime desc",
                            top=top if top <= 300 else None,
-                           filter=filter_string if filter_string and len(filter_string) > 0 else None)
+                           filter=filter_string if filter_string else None)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/custom.py]: maybe `filter_string or None`?

File: src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/__init__.py [hunk 2]
```diff
@@ -7,8 +7,8 @@
 
 
 def load_params(_):
-    import azure.cli.command_modules.profile._params  # pylint: disable=redefined-outer-name
+    import azure.cli.command_modules.profile._params  # pylint: disable=redefined-outer-name, unused-variable
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/__init__.py]: Why we need this extra suppression?

**Comment by troydai** [comment]
Thanks for the quick turnaround!

**Pull request merged**